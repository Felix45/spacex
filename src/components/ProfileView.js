import React from 'react';
import { useSelector } from 'react-redux';
import ListGroup from 'react-bootstrap/ListGroup';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';

const ProfileView = () => {
  const { rockets } = useSelector((state) => state.rockets);
  const { items } = useSelector((state) => state.missions);

  const filteredRockets = rockets.filter((rocket) => !!rocket.booked);
  const missions = items.filter((mission) => mission.reserved === true);

  return (
    <Row>
      <Col md={6}>
        <h2>My Missions</h2>
        <ListGroup>
          {
            missions.map((mission) => (
              <ListGroup.Item key={mission.mission_id}>{mission.mission_name}</ListGroup.Item>
            ))
          }
        </ListGroup>
      </Col>

      <Col md={6}>
        <h2>My Rockets</h2>
        <ListGroup>
          {
            filteredRockets.map((rocket) => (
              <ListGroup.Item key={rocket.id}>{ rocket.rocket_name }</ListGroup.Item>
            ))
          }
        </ListGroup>
      </Col>
    </Row>
  );
};

export default ProfileView;
