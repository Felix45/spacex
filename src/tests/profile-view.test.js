import React from 'react';
import ProfileView from '../components/ProfileView';
import renderWithProviders from './test-utils';

describe('ProfileView component', () => {
  it('Renders correctly', () => {
    const { asFragment } = renderWithProviders(<ProfileView />);
    const firstRender = asFragment();

    expect(firstRender).toMatchSnapshot();
  });
});
