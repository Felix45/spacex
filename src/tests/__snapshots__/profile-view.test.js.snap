// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileView component Renders correctly 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <div
      class="col-md-6"
    >
      <h2>
        My Rockets
      </h2>
      <div
        class="list-group"
      />
    </div>
  </div>
</DocumentFragment>
`;
